src/window.rs

```diff
@@ -0,0 +1,149 @@
+//! Displays an image in a window created by sdl2.
+
+use image::RgbaImage;
+use sdl2::event::{WindowEvent, Event};
+use sdl2::rect::Rect;
+use sdl2::keyboard::Keycode;
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::surface::Surface;
+
+/// Displays the provided RGBA image in a new window.
+pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_height: u32) {
```

Comment:
It would be nice if this was generic over pixel type, but I'm happy to merge without that.

Please mention the min window size here so that no-one gets confused when they try to draw a 100x100 window and it doesn't work.

Reply:
I wanted to make the function generic too, however, the function becomes a lot more complex, because inevitably the input image has to be converted to an RGB image. I figured it would be easier and simpler for the user to call convert from the ConvertBuffer trait instead of having to figure out what kind of image is being passed into the function and correctly converting it to RGB.